docs: Add module development guide for west (#112656)
zephyrproject-rtos/zephyr: doc/develop/west/built-in.rst

   - in all other projects where no rebase or merge is needed it keeps
     your branches in place.

**Developing a project:**

When developing in a west-managed project, create a local branch before
committing your work. A plain ``west update`` will leave that branch and its

> commits unchanged, then check out the manifest revision as a detached ``HEAD``.

Reviewer: Would not use "then" for saying that something got done after *not* doing something:

```suggestion
commits unchanged, and checks out the manifest revision as a detached ``HEAD``.
```

Commit or stash local changes before running ``west update``.

> #. Run ``west update`` to synchronize the workspace before starting work.

Reviewer: ```suggestion
#. Run ``west update`` to synchronize the workspace before starting your work.
```

Author: Thank you for the thoughtful review and your patience in explaining the concerns. I agree this change isn't worth pursuing, so I'm closing the PR.

#. Change to the project directory and create a local branch:

   .. code-block:: shell

      cd modules/lib/zcbor

>       git checkout -b my-feature

Reviewer: Would recommend to use the more modern `git switch` instead of `git checkout`:

```suggestion
      git switch -c my-feature
```

#. Make your changes and commit them:

   .. code-block:: shell

      git add <files>
      git commit -s

#. When updating the workspace later, run ``west update`` and check out your
   branch again:

   .. code-block:: shell

      west update

>       git checkout my-feature

Reviewer: ```suggestion
      git switch my-feature
```
